jetty-io/src/test/java/org/eclipse/jetty/io/SocketChannelEndPointOpenCloseTest.java

```diff
@@ -58,9 +61,51 @@ public static void close() throws Exception
     private EndPointPair newConnection() throws Exception
     {
         EndPointPair c = new EndPointPair();
-
-        c.client = new SocketChannelEndPoint(SocketChannel.open(connector.socket().getLocalSocketAddress()), null, null, null);
-        c.server = new SocketChannelEndPoint(connector.accept(), null, null, null);
+        SelectionKey k = new SelectionKey()
+        {
+            @Override
+            public SelectableChannel channel()
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public Selector selector()
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public boolean isValid()
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void cancel()
+            {
+            }
+
+            @Override
+            public int interestOps()
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public SelectionKey interestOps(int ops)
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public int readyOps()
+            {
+                throw new UnsupportedOperationException();
+            }
+        };
+        c.client = new SocketChannelEndPoint(SocketChannel.open(connector.socket().getLocalSocketAddress()), null, k, null);
+        c.server = new SocketChannelEndPoint(connector.accept(), null, k, null);
```

Comment:
Oh, no! 😞 
Aren't unit/mock/whatever tests great? Not!

Would it be possible to rewrite this test to something less hacky that actually tests real code?

Reply:
I'm not sure of what you mean. This test is checking the open/shutdown state of `SocketChannelEndPoint`, which happened to not need the selection key but now does. Why would we want to change the scope of this test because we changed an implementation detail?